src/controller/BoardController.java

```diff
@@ -32,13 +33,12 @@ public void actionPerformed(ActionEvent e) {
         String plantType = plantSelectionPanel.getPlant();
 
         if (plantType == PlantTypes.SUNFLOWER.getName()) {
-            if (Sunflower.getCost() <= this.statsPanel.getStats().getSunPoints()) {
-                Sunflower sunflower = new Sunflower();
-            	if (this.board.placePlant(sunflower, this.row, this.col) == false) {
-            		JOptionPane.showMessageDialog(null, "There's already a plant there.\nChoose a different location!");	
-            	} else {
+            Sunflower sunflower = new Sunflower();
+            if (sunflower.getSunPointCost() <= this.statsPanel.getStats().getSunPoints()) {
+                if (this.board.placePlant(sunflower, this.row, this.col) == true) {
+                    this.statsPanel.getStats().increaseSunPointsGenerationRate(sunflower.getSunPointGeneratedPerTurn());
                     this.statsPanel.update(sunflower.getSunPointCost());
```

Comment:
You should not have to call update. The model should notify all the listeners (views) when something changes.
Model -> View -> Controller.